internal/ingress/annotations/authtls/main.go

```diff
@@ -42,7 +42,7 @@ const (
 
 var (
 	authVerifyClientRegex = regexp.MustCompile(`^(on|off|optional|optional_no_ca)$`)
-	redirectRegex         = regexp.MustCompile(`^((https?://)?[A-Za-z0-9\-.]*(:\d+)?/[A-Za-z0-9\-.]*)?$`)
+	redirectRegex         = regexp.MustCompile(`^(?:https?://)?(?:\b\w+\b)?(?:/\w+)*/?$`)
```

Comment:
Can you explain the use of `?:` here? As far as I can tell we are not capturing anything here, just checking if the value matches.

Also `\w` is not the same as `[A-Za-z0-9\-.]`. The former does not include dashes nor periods, it only matches `[a-zA-Z0-9_]`. Additionally the new RegEx does not cover ports in the host part.